daml-lf/transaction/src/main/scala/com/daml/lf/TransactionNodeStatistics.scala

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf
+package transaction
+
+object TransactionNodeStatistics {
+
+  /** Container for transaction statistics.
+    *
+    * @param creates number of creates nodes,
+    * @param consumingExercisesByCid number of consuming exercises by contract ID nodes,
+    * @param nonconsumingExercisesByCid number of non-consuming Exercises by contract ID nodes,
+    * @param consumingExercisesByKey number of consuming exercise by contract key nodes,
+    * @param nonconsumingExercisesByKey number of non-consuming exercise by key nodes,
+    * @param fetchesByCid number of fetch by contract ID nodes,
+    * @param fetchesByKey number of fetch by key nodes,
+    * @param lookupsByKey number of lookup by key nodes,
+    * @param rollbacks number of rollback nodes.
+    */
+  final case class Actions(
+      creates: Int,
+      consumingExercisesByCid: Int,
+      nonconsumingExercisesByCid: Int,
+      consumingExercisesByKey: Int,
+      nonconsumingExercisesByKey: Int,
+      fetchesByCid: Int,
+      fetchesByKey: Int,
+      lookupsByKey: Int,
+  ) {
+
+    def +(that: Actions) =
+      Actions(
+        creates = this.creates + that.creates,
+        consumingExercisesByCid = this.consumingExercisesByCid + that.consumingExercisesByCid,
+        nonconsumingExercisesByCid =
+          this.nonconsumingExercisesByCid + that.nonconsumingExercisesByCid,
+        consumingExercisesByKey = this.consumingExercisesByKey + that.consumingExercisesByKey,
+        nonconsumingExercisesByKey =
+          this.nonconsumingExercisesByKey + that.nonconsumingExercisesByKey,
+        fetchesByCid = this.fetchesByCid + that.fetchesByCid,
+        fetchesByKey = this.fetchesByKey + that.fetchesByKey,
+        lookupsByKey = this.lookupsByKey + that.lookupsByKey,
+      )
+
+    def exercisesByCid: Int = consumingExercisesByCid + nonconsumingExercisesByCid
+    def exercisesByKey: Int = consumingExercisesByKey + nonconsumingExercisesByKey
+    def exercises: Int = exercisesByCid + exercisesByKey
+    def consumingExercises: Int = consumingExercisesByCid + consumingExercisesByKey
+    def nonconsumingExercises: Int = nonconsumingExercisesByCid + nonconsumingExercisesByKey
+    def fetches: Int = fetchesByCid + fetchesByKey
+    def byKeys: Int = exercisesByKey + fetchesByKey + lookupsByKey
+    def actions: Int = creates + exercises + fetches + lookupsByKey
+    def nodes: Int = actions + rollbacks
```

Comment:
```suggestion
```
I don’t think this even compiles given that you deleted `rollbacks`

Reply:
done